common/ayon_common/distribution/control.py

```diff
@@ -1368,24 +1368,35 @@ def _post_source_process(
             yield False
             return
 
+        yield None
+
         # Move unzipped content to target directory
         moved_paths = []
+        rename_attemps = 5
         for subname in os.listdir(unzip_dirpath):
             target_path = os.path.join(self.target_dirpath, subname)
             src_path = os.path.join(unzip_dirpath, subname)
             dst_path = os.path.join(self.target_dirpath, subname)
-            try:
-                os.rename(src_path, dst_path)
-            except Exception:
-                message = (
-                    "Failed to move unzipped content to target directory."
-                )
-                source_progress.set_failed(message)
-                self.log.warning(
-                    f"{self.item_label}: {message}",
-                    exc_info=True
-                )
-                failed = True
+            self.log.debug("Copying %s to %s", src_path, dst_path)
+            for attempt in range(rename_attemps):
+                try:
+                    os.rename(src_path, dst_path)
+                    break
+                except Exception:
+                    if attempt < (rename_attemps - 1):
+                        yield None
+                        continue
+                    message = (
+                        "Failed to move unzipped content to target directory."
+                    )
+                    source_progress.set_failed(message)
+                    self.log.warning(
+                        f"{self.item_label}: {message}",
```

Comment:
For sake of debugging - should we log the exception. E.g. an `OSError` may occur if `dst_path` already exists which is not what we're checking for explicitly here?